lib/domain/usecase/room/update_pinned_messages_interactor.dart

```diff
@@ -14,10 +14,16 @@ class UpdatePinnedMessagesInteractor {
     try {
       yield Right(UpdatePinnedEventsInitial());
       final currentPinnedEvents = room.pinnedEventIds.toSet();
-      if (action == PinnedMessagesActionEnum.pin) {
-        currentPinnedEvents.addAll(eventIds);
-      } else {
-        currentPinnedEvents.removeAll(eventIds);
+      switch (action) {
+        case PinnedMessagesActionEnum.pin:
+          currentPinnedEvents.addAll(eventIds);
+          break;
+        case PinnedMessagesActionEnum.unpin:
+          currentPinnedEvents.removeAll(eventIds);
+          break;
+        case PinnedMessagesActionEnum.unpinAll:
+          currentPinnedEvents.clear();
+          break;
       }
```

Comment:
Pls check it again, naming is not clean for me

Reply:
I don't get what is not clean, what name do you propose

Reply:
If the action is unpinned then why use `removeAll`. I mean `removeAll` will remove all pinned events

Reply:
it's the name of the list function, removeAll takes a list of element to remove from the list

Reply:
IMO: Change `removeAll ` => `unpin`

Reply:
https://api.flutter.dev/flutter/dart-core/Set/removeAll.html